src/table/boot.rs

```diff
@@ -646,6 +646,15 @@ pub enum MemoryType: u32 => {
     PERSISTENT_MEMORY       = 14,
 }}
 
+impl MemoryType {
+    /// Construct a custom `MemoryType`. Values in the range `0x80000000..=0xffffffff` are free for use if you are
```

Comment:
What would be the usefulness of using a custom memory type?

Reply:
I use them to track allocations I ask UEFI to make for specific purposes, such as for the kernel's heap vs. extra images loaded by the bootloader. These can then be used when processing the UEFI memory map to make the memory map passed to the kernel. You can see an example [here](https://github.com/IsaacWoods/pebble/blob/master/kernel/efiloader/src/main.rs#L324).

Reply:
Alright, it sounds reasonable.